fuel/src/test/kotlin/com/github/kittinunf/fuel/RequestHeaderTest.kt

```diff
@@ -46,4 +47,22 @@ class RequestHeaderTest : BaseTestCase() {
         assertThat(string, containsString(headerValue))
     }
 
+    @Test
+    fun multipleHeadersByTheSameKeyWillBeCorrectlyFormatted(){
+        val request = manager.request(Method.GET, "/get")
+                .header("foo" to "bar","a" to "b", "cookie" to "val1=x", "cookie" to "val2=y","cookie" to "val3=z", "cookie" to "val4=j")
+
+        assertEquals("[ val1=x,val2=y,val3=z,val4=j ]",request.headers["cookie"] )
+        assertEquals("bar",request.headers["foo"] )
+        assertEquals("b",request.headers["a"] )
+    }
+
+    @Test
+    fun multipleHeadersByTheSameKeyWillShowLastUsingMap(){
+        val request = manager.request(Method.GET, "/get")
+                .header(mapOf("cookie" to "val1=x", "cookie" to "val2=y"),false)
```

Comment:
Is this because `mapOf` does that? Probably should need an entry in the usage on `README.md`. Multiple values for a header? This is how you do it:

Reply:
@SleeplessByte 
The problem with map is, (as I am sure you are aware), when you add multiple values for a key, they are all over written by the last one to write

Reply:
Yep! So we/you should document that, as it's "logical" if you realise it :)

Reply:
@SleeplessByte how would you suggest we document this?

Reply:
**Adding multiple values to a header**
If you want to mimic multiple values of a header such as:
```
Cookie: foo=bar
Cookie: baz=42
```

You can use:
```Kotlin
.header("Cookie" to "foo=bar", "Cookie" to "baz=42")
// => Cookie: foo=bar; baz=42
```

It is important to directly call `.header` and that you don't wrap the arguments with `mapOf`, because `mapOf` will simply overwrite any previously set value:

```Kotlin
.header(mapOf("Cookie" to "foo=bar", "Cookie" to "baz=42"))
// => Cookie: baz=42

Reply:
I will add this to the readme and shall I add kdocs?

Reply:
That would be nice! I traverse source code often myself 🥉